src/helpers/OptionList.h

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2023, Nexus6 <nexus6.haiku@icloud.com>
+ * All rights reserved. Distributed under the terms of the MIT license.
+ */
+#pragma once
+
+
+#include <MenuField.h>
+#include <MenuItem.h>
+#include <Messenger.h>
+#include <ObjectList.h>
+#include <PopUpMenu.h>
+#include <SupportDefs.h>
+
+#include <iterator>
+
+#include "GException.h"
+#include "GMessage.h"
+#include "Log.h"
+#include "Utils.h"
+
+namespace Genio::UI {
+
+	template <typename T>
+	  struct is_iterator {
+	  static char test(...);
+
+	  template <typename U,
+		typename=typename std::iterator_traits<U>::difference_type,
+		typename=typename std::iterator_traits<U>::pointer,
+		typename=typename std::iterator_traits<U>::reference,
+		typename=typename std::iterator_traits<U>::value_type,
+		typename=typename std::iterator_traits<U>::iterator_category
+	  > static long test(U&&);
+
+	  constexpr static bool value = std::is_same<decltype(test(std::declval<T>())),long>::value;
+	};
+
+	template <typename T>
+	class OptionList : public BMenuField {
+	public:
+		OptionList(const char* name,
+			const char* label,
+			const char* text,
+			bool fixed_size = true,
+			uint32 flags = B_WILL_DRAW | B_NAVIGABLE)
+			:
+			BMenuField(name, label, fMenu = new BPopUpMenu(text), fixed_size, flags),
+			fMessenger(nullptr)
+		{
+			fMenu->SetLabelFromMarked(true);
+		}
+
+		~OptionList()
+		{
+			delete fMessenger;
+		}
+
+		void MakeEmpty()
+		{
+			for (int32 index = fMenu->CountItems() - 1; index > -1; index--)
+				fMenu->RemoveItem(index);
+		}
+
+
+		void AddItem(BString &name, T value, uint32 command,
+			bool invokeItemMessage = true, bool marked = false)
+		{
+			status_t status;
+			auto message = new BMessage(command);
+			if constexpr (std::is_pointer<T>::value) {
+				status = message->AddPointer("value", value);
+			} if constexpr (std::is_same<BString, T>::value) {
```

Comment:
Is this correct ? Shouldn't this be an "else if", maybe ?

Reply:
absolutely, yes.